egui/src/widgets/button.rs

```diff
@@ -242,26 +242,34 @@ impl<'a> Widget for Checkbox<'a> {
         let total_extra = button_padding + vec2(icon_width + icon_spacing, 0.0) + button_padding;
 
         let wrap_width = ui.available_width() - total_extra.x;
-        let text = text.into_galley(ui, None, wrap_width, TextStyle::Button);
 
-        let mut desired_size = total_extra + text.size();
-        desired_size = desired_size.at_least(spacing.interact_size);
+        let mut desired_size = total_extra;
+        let text = if !text.text().is_empty() {
+            let text = text.into_galley(ui, None, wrap_width, TextStyle::Button);
+            desired_size += text.size();
+            desired_size = desired_size.at_least(spacing.interact_size); //Here I'm unsure - what is interacting size?
```

Comment:
From [the docs](https://docs.rs/egui/latest/egui/style/struct.Spacing.html#structfield.interact_size):

> Minimum size of a DragValue, color picker button, and other small widgets. interact_size.y is the default height of button, slider, etc. Anything clickable should be (at least) this size.

The original idea was that it is bad to have very small buttons, because they are difficult to click (especially on touch screens), but in this case you presumably want a small button.

`desired_size = desired_size.at_least(Vec2::splat(spacing.interact_size.y));` could be a good workaround, and should be applied whether or not there is text.

Reply:
I've removed the original 'at_least' instruction, and replaced it with your suggestion. Is this what you had in your mind?
Or an if/else - if text is empty, then your suggestion, else original code?